processor/swok8sworkloadtypeprocessor/processor.go

```diff
@@ -0,0 +1,224 @@
+	resourceMetrics := md.ResourceMetrics()
+	for rmIdx := range resourceMetrics.Len() {
+		rm := resourceMetrics.At(rmIdx)
+		scopeMetrics := rm.ScopeMetrics()
+		for smIdx := range scopeMetrics.Len() {
+			sm := scopeMetrics.At(smIdx)
+			metrics := sm.Metrics()
+			for mIdx := range metrics.Len() {
+				m := metrics.At(mIdx)
+				switch m.Type() {
+				case pmetric.MetricTypeGauge:
+					datapoints := m.Gauge().DataPoints()
+					for dpIdx := range datapoints.Len() {
+						dp := datapoints.At(dpIdx)
+						err := cp.setWorkloadTypeForGivenDatapointAttributes(dp.Attributes())
+						if err != nil {
+							return pmetric.Metrics{}, fmt.Errorf("error getting workload type for metric '%s': %w", m.Name(), err)
+						}
+					}
+				case pmetric.MetricTypeSum:
+					datapoints := m.Sum().DataPoints()
+					for dpIdx := range datapoints.Len() {
+						dp := datapoints.At(dpIdx)
+						err := cp.setWorkloadTypeForGivenDatapointAttributes(dp.Attributes())
+						if err != nil {
+							return pmetric.Metrics{}, fmt.Errorf("error getting workload type for metric '%s': %w", m.Name(), err)
+						}
+					}
+				case pmetric.MetricTypeHistogram:
+					datapoints := m.Histogram().DataPoints()
+					for dpIdx := range datapoints.Len() {
+						dp := datapoints.At(dpIdx)
+						err := cp.setWorkloadTypeForGivenDatapointAttributes(dp.Attributes())
+						if err != nil {
+							return pmetric.Metrics{}, fmt.Errorf("error getting workload type for metric '%s': %w", m.Name(), err)
+						}
+					}
+				case pmetric.MetricTypeExponentialHistogram:
+					datapoints := m.ExponentialHistogram().DataPoints()
+					for dpIdx := range datapoints.Len() {
+						dp := datapoints.At(dpIdx)
+						err := cp.setWorkloadTypeForGivenDatapointAttributes(dp.Attributes())
+						if err != nil {
+							return pmetric.Metrics{}, fmt.Errorf("error getting workload type for metric '%s': %w", m.Name(), err)
+						}
+					}
+				case pmetric.MetricTypeSummary:
+					datapoints := m.Summary().DataPoints()
+					for dpIdx := range datapoints.Len() {
+						dp := datapoints.At(dpIdx)
+						err := cp.setWorkloadTypeForGivenDatapointAttributes(dp.Attributes())
+						if err != nil {
+							return pmetric.Metrics{}, fmt.Errorf("error getting workload type for metric '%s': %w", m.Name(), err)
+						}
+					}
+				default:
+					cp.logger.Debug("unsupported metric type", zap.Any("metricType", m.Type()))
+				}
+			}
+		}
+	}
+
+	return md, nil
+}
+
+func (cp *swok8sworkloadtypeProcessor) Start(ctx context.Context, _ component.Host) error {
+	cancelCtx, cancelFunc := context.WithCancel(context.Background())
+	cp.cancel = cancelFunc
+
+	client, err := cp.config.getK8sClient()
+	if err != nil {
+		return err
+	}
+	cp.k8sClient = client
+
+	cp.factory = informers.NewSharedInformerFactory(cp.k8sClient, cp.config.WatchSyncPeriod)
+
+	registerWorkloadInformer := func(workloadType K8sWorkloadType, kind string, informer cache.SharedIndexInformer) {
+		cp.informers[workloadType] = informer
```

Comment:
It is common practices that startup of these kind of instances are logged, so I would add here something like `cp.logger.Info("Informer started", zap.String("workloadType", string(workloadType)))` so that we see in collector log how many informers are actually running.

Reply:
I had the logging already implemented locally, but I was waiting with it (and unit tests) for an initial review of the processor's concept first.